packages/xstate-graph/src/types.ts

```diff
@@ -1,9 +1,64 @@
-import { State, EventObject, StateValue } from 'xstate';
+import {
+  State,
+  EventObject,
+  StateValue,
+  StateNode,
+  TransitionDefinition
+} from 'xstate';
 
 export interface TransitionMap {
   state: StateValue | undefined;
 }
 
+export type JSONSerializable<T extends object, U> = T & {
+  toJSON: () => U;
+};
+
+export type DigraphLabel = JSONSerializable<
+  {
+    [key: string]: any;
```

Comment:
we are the only producers of those graphs so far, do we need to allow indexer properties? they make the consuming code more fragile to mistakes

Reply:
Yes, in some layout algorithms, label nodes can contain width/height and other meta data. This shouldn't be restricted.

Reply:
That would mean that people would have to attach properties manually after calling `toGraph` or they would have to provide compatible graph structures (containing extra metadata) to us - right now we have no tools/utils for consuming those. We only produce them - so the question is: we have to include this already? If we want to make this type-safe we could also accept extra `Extension` generic for those types, rather than allowing yolo indexers 😅

Reply:
That's fair, I'll remove it for now.